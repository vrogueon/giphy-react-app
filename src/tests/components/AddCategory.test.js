import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme/build';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
       
        const input = wrapper.find( 'input' ) ;

        const value = 'Hola mundo';

       input.simulate('change', { target: { value } });
       expect( wrapper.find('p').text().trim() ).toBe(value);

    });
    
    test('No debe de postear la informacion on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
       
       const value = 'test';
       
       wrapper.find('input').simulate('change', { target: { value } });
       wrapper.find('form').simulate('submit', { preventDefault(){} });

       expect( setCategories ).toHaveBeenCalled();
       expect(wrapper.find('input').prop('value')).toBe('');
    });
    
    
});
