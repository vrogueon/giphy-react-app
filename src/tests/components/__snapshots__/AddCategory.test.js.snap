// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <AddCategory/> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
